templates/front/my_checks_desktop.html

```diff
@@ -12,6 +12,7 @@
         <th></th>
     </tr>
     {% for check in checks %}
+    {% if check.id in allowed_checks or check.user_id == user.id %}
```

Comment:
The concern here is the O(n^2) op happening in the view. FWIW, if we have 1 million `checks`, checking that each is contained in `allowed_checks` will be rather slow. The first optimization we can do here is write the condition this way `{% if check.user_id == user.id or check.id in allowed_checks %}`